plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/schema/confluent/ConfluentSchemaRegistryTableDescriptionSupplier.java

```diff
@@ -166,21 +171,23 @@ private String resolveSubject(String candidate)
         TopicAndSubjects topicAndSubjects = parseTopicAndSubjects(schemaTableName);
 
         String tableName = topicAndSubjects.getTableName();
-        Optional<TopicAndSubjects> topicAndSubjectsFromCache = Optional.ofNullable(topicAndSubjectsSupplier.get().get(tableName));
-
-        // Use the topic from cache, if present, in case the topic is mixed case
-        String topic = topicAndSubjectsFromCache.map(TopicAndSubjects::getTopic).orElse(topicAndSubjects.getTopic());
-        Optional<String> keySubject = topicAndSubjects.getKeySubject()
-                .or(() -> topicAndSubjectsFromCache.flatMap(TopicAndSubjects::getKeySubject));
-        Optional<String> valueSubject = topicAndSubjects.getValueSubject()
-                .or(() -> topicAndSubjectsFromCache.flatMap(TopicAndSubjects::getValueSubject));
+        if (topicAndSubjectsSupplier.get().containsKey(tableName)) {
+            // Use the topic from cache, if present, in case the topic is mixed case
+            Collection<TopicAndSubjects> topicAndSubjectsCollection = topicAndSubjectsSupplier.get().get(tableName);
+            checkState(topicAndSubjectsCollection.size() == 1, "Ambiguous topics " + topicAndSubjectsCollection.stream().map(TopicAndSubjects::getTopic).collect(joining(",")));
```

Comment:
this should be a `TrinoException` as this error is not a developer mistake, but a situation that could happen in the wild.